wrapper/src/main/java/software/amazon/jdbc/LeastConnectionsHostSelector.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc;
+
+import com.zaxxer.hikari.HikariDataSource;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import software.amazon.jdbc.util.Messages;
+import software.amazon.jdbc.util.SlidingExpirationCache;
+
+public class LeastConnectionsHostSelector implements HostSelector {
+  // This method should not be called.
+  @Override
+  public HostSpec getHost(List<HostSpec> hosts, HostRole role) throws SQLException {
+    throw new SQLException(Messages.get("HostSelector.leastConnectionsIncorrectGetHost"));
+  }
+
+  public HostSpec getHost(
```

Comment:
I feel like this breaks/circumvents the point of using the HostSelect interface. Have you considered providing `databasePools` through the constructor instead?